pkg/daemon/daemon.go

```diff
@@ -634,16 +637,16 @@ func (dn *Daemon) Run(stopCh <-chan struct{}, exitCh <-chan error) error {
 
 	go wait.Until(dn.worker, time.Second, stopCh)
 
-	for {
```

Comment:
The `for` loop is superfluous since. Once we get the `sigterm` we _should_ finish the work and then get out of the way since a `sigkill` could/likely follow.